### PR TITLE
systemd: set up environment in rbdmap unit file

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/14984 Fixes: #14984

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
The bug report says "/usr/bin/rbdmap requires the RBDMAPFILE variable to be set," which doesn't sound right. Shouldn't the code default to a sensible value if the environment variable is not set?

**Comment by smithfarm** [comment]
@tchaikov @branto1

**Comment by smithfarm** [comment]
I looked in `qa/workunits/` but did not see any obvious evidence that tests are exercising rbdmap. Any idea what suite I should run to test this?

**Comment by smithfarm** [comment]
Spawning a cluster for manual testing with: `cw ceph-qa-suite --simultaneous-jobs 9 --verbose --teuthology-git-url http://github.com/dachary/teuthology --teuthology-branch openstack --ceph-qa-suite-git-url http://github.com/dachary/ceph-qa-suite --suite-branch wip-ceph-disk --ceph-git-url http://github.com/SUSE/ceph --ceph wip-14984 --suite ceph-disk --filter centos_7`

**Comment by smithfarm** [comment]
Scope of test: on the cluster deployed with packages built from this PR, run `systemctl start rbdmap.service` and inspect the logs for bogus messages. Add `set -x` to `/usr/bin/rbdmap` and restart the service. Confirm via `journalctl` that the `/etc/ceph/rbdmap` file is processed. Then, back out the changes and confirm that bogus messages appear in the log and `/etc/ceph/rbdmap` file is not processed.

The test scope does not extend to creating an RBD image and using `rbdmap` to map it to a device and mount it.

Test result: **passed**

**Comment by smithfarm** [comment]
**running** http://149.202.190.143:8081/ubuntu-2016-03-20_17:35:19-rbd:singleton:all:rbdmap_RBDMAPFILE.yaml-wip-14984---basic-openstack/

`cw ceph-qa-suite --simultaneous-jobs 9 --verbose --teuthology-git-url http://github.com/dachary/teuthology --teuthology-branch openstack --ceph-qa-suite-git-url http://github.com/smithfarm/ceph-qa-suite --suite-branch wip-14984 --ceph-git-url http://github.com/SUSE/ceph --ceph wip-14984 --suite rbd/singleton/all/rbdmap_RBDMAPFILE.yaml`

**Comment by smithfarm** [comment]
Running the workunit as a standalone test against the jewel branch to demonstrate that the workunit fails when the bug is present.

*running* http://149.202.190.143:8081/ubuntu-2016-03-20_21:54:12-rbd:singleton:all:rbdmap_RBDMAPFILE.yaml-jewel---basic-openstack/

`cw ceph-qa-suite --simultaneous-jobs 9 --verbose --teuthology-git-url http://github.com/dachary/teuthology --teuthology-branch openstack --ceph-qa-suite-git-url http://github.com/smithfarm/ceph-qa-suite --suite-branch wip-14984 --ceph-git-url http://github.com/SUSE/ceph --ceph jewel --suite rbd/singleton/all/rbdmap_RBDMAPFILE.yaml distros/all/centos_7.2.yaml`

**Comment by smithfarm** [comment]
Running the workunit as a standalone test against **this PR** to demonstrate that the workunit succeeds.

**running** http://149.202.190.143:8081/ubuntu-2016-03-20_22:01:27-rbd:singleton:all:rbdmap_RBDMAPFILE.yaml-wip-14984---basic-openstack/

`cw ceph-qa-suite --simultaneous-jobs 9 --verbose --teuthology-git-url http://github.com/dachary/teuthology --teuthology-branch openstack --ceph-qa-suite-git-url http://github.com/smithfarm/ceph-qa-suite --suite-branch wip-14984 --ceph-git-url http://github.com/SUSE/ceph --ceph wip-14984 --suite rbd/singleton/all/rbdmap_RBDMAPFILE.yaml distros/all/centos_7.2.yaml`

**Comment by smithfarm** [comment]
Associated `ceph/ceph-qa-suite` PR: https://github.com/ceph/ceph-qa-suite/pull/902

File: qa/workunits/rbd/test_rbdmap_RBDMAPFILE.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# When the bug is present, starting the rbdmap service causes
+# a bogus log message to be emitted to the log because the RBDMAPFILE
+# environment variable is not set.
+#
+# When the bug is not present, starting the rbdmap service will emit 
+# no log messages, because /etc/ceph/rbdmap does not contain any lines 
+# that require processing.
+#
+set -ex
+
```
Comment (smithfarm) [inline qa/workunits/rbd/test_rbdmap_RBDMAPFILE.sh]: here we could add a packaging test that would check `which rbdmap`

**Comment by branto1** [comment]
lgtm

**Comment by smithfarm** [comment]
@branto1 Thanks for the review! Could you take a look at #8224 as well?

**Pull request merged**